### PR TITLE
Surface supported interaction profiles from OpenXR runtime

### PR DESCRIPTION
**Pull request by msub2**
This introduces support for a wider range of OpenXR interaction profiles and properly handles updating the associated XRInputSource when the active interaction profile changes.

### CODE REVIEW

**Comment by msub2** [comment]
Leaving this as a draft right now while I wait on some word from [ralith/openxrx#228](https://github.com/Ralith/openxrs/issues/165)

File: webxr/openxr/input.rs [hunk 1]
```diff
@@ -190,32 +194,35 @@ impl OpenXRInput {
...
-                .get_bindings(instance, "trigger/value", Some("squeeze/click"))
-                .into_iter(),
-        );
-        let path_controller = instance
-            .string_to_path("/interaction_profiles/microsoft/motion_controller")
-            .unwrap();
-        instance
-            .suggest_interaction_profile_bindings(path_controller, &bindings)
-            .unwrap();
+        INTERACTION_PROFILES.iter().for_each(|profile| {
```
Comment (jdm) [inline webxr/openxr/input.rs:198]: Why use for_each over a for loop here?

File: webxr/openxr/interaction_profiles.rs [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+    META_TOUCH_PRO_CONTROLLER_PROFILE,
+    META_TOUCH_PLUS_CONTROLLER_PROFILE,
+    META_TOUCH_CONTROLLER_RIFT_CV1_PROFILE,
+    META_TOUCH_CONTROLLER_QUEST_1_RIFT_S_PROFILE,
+    META_TOUCH_CONTROLLER_QUEST_2_PROFILE,
+    SAMSUNG_ODYSSEY_CONTROLLER_PROFILE,
+    VALVE_INDEX_CONTROLLER_PROFILE,
+];
+
+pub fn get_profiles_from_path(path: String) -> Vec<&'static str> {
```
Comment (jdm) [inline webxr/openxr/interaction_profiles.rs:360]: Nit: this could return a static slice instead of a Vec, right?

File: webxr/openxr/input.rs [hunk 3]
```diff
@@ -190,32 +194,35 @@ impl OpenXRInput {
...
+            if let Some(extension_name) = profile.required_extension {
+                if !supported_interaction_profiles
+                    .contains(&String::from_utf8(extension_name.to_vec()).unwrap())
```
Comment (jdm) [inline webxr/openxr/input.rs]: I think it should be possible to use str::from_utf8 and avoid the to_vec and String allocations.

File: webxr/openxr/input.rs [hunk 4]
```diff
@@ -190,32 +194,35 @@ impl OpenXRInput {
...
+            let mut bindings = right_hand.get_bindings(instance, select, squeeze);
+            bindings.extend(
+                left_hand
+                    .get_bindings(instance, select, squeeze)
+                    .into_iter(),
+            );
+            let path_controller = instance
+                .string_to_path(profile.path)
+                .expect(format!("Invalid interaction profile path: {}", profile.path).as_str());
+            let _ = instance
```
Comment (jdm) [inline webxr/openxr/input.rs:215]: Should this be an if let/else (or a match) instead of a map_err?

**Review by jdm — COMMENTED** [review]
Looking good so far!

**Comment by msub2** [comment]
ah whoops there's a typo in the touch plus profile still, fixing

**Review by jdm — APPROVED** [review]

**Pull request merged**